lib/src/identity_codec.dart

```diff
@@ -0,0 +1,29 @@
+import 'dart:convert';
+
+class _IdentityConverter<T> extends Converter<T, T> {
+  @override
+  T convert(T input) => input;
+}
+
+/// A [Codec] that converts from T to T doing absolutely nothing.
+///
+/// This is the identity codec, the one that passes input to output when
+/// decoding and encoding. This codec is mostly useful when implementing generic
+/// code that can be fused with a [Codec] and you need an identity codec for the
+/// base case.
+///
+/// Note, that when fused with another [Codec] the identity codec disppears.
+class IdentityCodec<T> extends Codec<T, T> {
+  const IdentityCodec();
+
+  @override
+  Converter<T, T> get decoder => _IdentityConverter();
```

Comment:
I prefer to be explicit instead of depending on inference, so consider `_IdentityConverter<T>()` here and below.

Reply:
Consider making it a field:
```dart
final Converter<T, T> decoder = _IdentityConverter<T>();
Converter<T, T> get encoder => decoder;
```
Then you only create one converter per codec. It costs more if you end up not using the codec, but otherwise it should be a wash.

Reply:
Neat trick..

To be fair I was hoping the JIT would just optimize it out...